lib/dependency-manager-adapters/workspace.js

```diff
@@ -56,24 +56,26 @@ module.exports = CoreObject.extend({
 
   changeToDependencySet(depSet) {
     // TODO: What should this do for tables? Nesting? Needs different output
-    this._packageAdapters.forEach(adapter => {
-      adapter.applyDependencySet(depSet);
-    });
-
-    return this._install(depSet).then(() => {
-      let deps = extend({}, depSet.dependencies || {}, depSet.devDependencies || {});
-      let currentDeps = Object.keys(deps).map((dep) => {
-        return {
-          name: dep,
-          versionExpected: deps[dep],
-          versionSeen: this._findCurrentVersionOf(dep),
-          packageManager: 'yarn',
-        };
+    let appliedDependencies = this._packageAdapters.map(adapter =>
+      adapter.applyDependencySet(depSet)
+    );
+
+    return RSVP.all(appliedDependencies).then(() => {
```

Comment:
Like above, this seems like a great fundamental fix. Are you certain that these fixes to properly entangle `applyDependencySet` aren't all that is needed to solve the issue you described?